uniffi_bindgen/src/bindings/python/templates/ObjectTemplate.py

```diff
@@ -14,17 +20,37 @@ def __del__(self):
             self._handle
         )
 
-    {% for cons in obj.alternate_constructors() -%}
+    # Used by alternative constructors or any methods which return this type.
+    # May return an existing instance, or create a new one if we can't find one.
     @classmethod
-    def {{ cons.name()|fn_name_py }}(cls, {% call py::arg_list_decl(cons) %}):
-        {%- call py::coerce_args_extra_indent(cons) %}
-        # Call the (fallible) function before creating any half-baked object instances.
-        handle = {% call py::to_ffi_call(cons) %}
+    def _get_or_make_instance_(cls, handle):
+        {# Look in our weak map for an instance already associated with this
+           pointer. If it exists we can return it, but the pointer we have
+           is a clone of an `Arc<>` so has a reference we need to destroy.
+        #}
```

Comment:
I feel like we probably want a mutex or other concurrency-control here around the updates to the identity map.